manifests/globals.pp

```diff
@@ -43,14 +43,18 @@
   }
 
   # Setup of the repo only makes sense globally, so we are doing it here.
+  # For Debian family if manage_package_repo is set at undef that not include mongodb::repo
+  # For other family if manage_package_repo is set at undef that include mongodb::repo
   case $facts['os']['family'] {
     'RedHat', 'Linux', 'Suse': {
-      class { 'mongodb::repo':
-        ensure              => present,
-        version             => pick($version, '3.6'),
-        use_enterprise_repo => $use_enterprise_repo,
-        repo_location       => $repo_location,
-        proxy               => $repo_proxy,
+      if $manage_package_repo != false {
```

Comment:
Instead of duplicate conditionals in the case statement, move the conditional it.

Reply:
That what I've doing in first step, but that create a breaking change. I need to type the variable for unsupported undef value, it's ok for that?

Also the default value change between os family and release, I thinking to create à hiera.yaml and data directory, any preference about the hierarchy ?

Reply:
Ah, I misunderstood the thing and failed to see that the variable can be `unset`.

I think that any resource in globals.pp / params.pp is a bad idea.  I think that $manage_* should always be booleans and not allow undef.  This sucks :unamused: 

Maybe move the comments I was wrong about just before the corresponding conditional (easy move for now, no major change)?  If you have the resources to do more refactoring, we could track it in another PR when this issue is fixed?

Reply:
Yes it's tsuck, that why I've write a comments x')

Ok I create another PR after this merge for set this variable as a boolean